serdaSimus/assignments/slidingwindow/lc53/lc53.js

```diff
@@ -1,7 +1,59 @@
+// Problem Statement #
+// Given an array of positive numbers and a positive number ‘k,’ find the maximum sum of any contiguous subarray of size ‘k’.
+
+/*
+  [] ... [1] ... [1,2,3,4,5]
+  I need accumulator: sum
+  I need var: largestValue
+  I need a counter: 
+
+  for loop: 
+    add each item to sum
+    check if counter >= k:
+      compare sum and largest: if sum is bigger:
+        i would save sum to larges, else I would continue
+        sum = sum - arr[counter - k]
+    counter++
+
+  return largestValue
+      
+*/
+// [1,2,3,4,6] k=3
+
 function max_subarray_size_k(k, arr) {
-  // TODO: Write your code here
+  // basic edge cases, k is null, arr is null,
+  // k is negative, k>arr.length
+  if (!arr.length || k > arr.length || k < 0 || k == null) return null;
```

Comment:
This is an amazing use of OR operator. Just remember that this can cause confusion for someone interviewing you who main language is not JS. This is a very opinionated topic, some will say it is better to write three/four quick if statements with implicit returns.

Final say: I like this and I think you should keep it. BUT, ask your interviewer "I decided to go with a very concise edge case here using a couple of OR statements. Are you comfortable reading this OR would you like me to break it down into seperate if statements" 

This will leave the power in their court^ If they do not understand what you wrote you now have a chance to clarify